modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/LimitNode.java

```diff
@@ -69,16 +79,17 @@ public void request(int rowsCnt) throws Exception {
             return;
         }
 
-        if (offset > 0 && rowsProcessed == 0) {
-            rowsCnt = offset + rowsCnt;
-        }
-
-        waiting = rowsCnt;
+        requested = rowsCnt;
```

Comment:
let's add assertion that previous requested count is not overridden

Reply:
done